core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -232,6 +232,12 @@ private[spark] class MapOutputTrackerMaster(conf: SparkConf)
   protected val mapStatuses = new TimeStampedHashMap[Int, Array[MapStatus]]()
   private val cachedSerializedStatuses = new TimeStampedHashMap[Int, Array[Byte]]()
 
+  // For each shuffleId we also maintain a Map from reducerId -> (location, size)
+  // Lazily populated whenever the statuses are requested from DAGScheduler
+  private val statusByReducer =
```

Comment:
nit: could you rename this to shuffleIdToReduceLocations? With the current name, I expected the main map to be keyed on reduce id.